src/swarm-doc/Swarm/Doc/Gen.hs

```diff
@@ -224,9 +229,37 @@ recipesToDot baseRobot classicTerm emap recipes = do
   mapM_ (uncurry (---<>)) (recipesToPairs recipeReqOut recipes)
   -- --------------------------------------------------------------------------
   -- also draw an edge for each entity that "yields" another entity
-  let yieldPairs = mapMaybe (\e -> (e ^. entityName,) <$> (e ^. entityYields)) . Map.elems $ entitiesByName emap
+  let yieldPairs = mapMaybe (\e -> (e ^. entityName,) <$> (e ^. entityYields)) . toList $ rgAllEntities graphData
   mapM_ (uncurry (.->.)) (both getE <$> yieldPairs)
```

Comment:
@byorgey I think you updated the graph display to Dot, but not the levels computation.

If Dot could lay out the graph well, the levels would not even be needed. But because we have so many connections, the levels at least group together nodes in order.